packages/protocol/src/utils/SPID.ts

```diff
@@ -13,6 +13,8 @@ import ValidationError from '../errors/ValidationError'
 
 type RequiredKeys<T, Keys extends keyof T> = Omit<T, Keys> & Required<Pick<T, Keys>>
 
+export type SPIDKey = string
```

Comment:
With template literal types (https://www.typescriptlang.org/docs/handbook/2/template-literal-types.html) we can actually do the following
```suggestion
export type SPIDKey = `${string}#${number}`
```

Reply:
Seems quite complicated to add that restriction as we have *many* input formats to support. If we can simplify `SPID.fromDefaults()`, it should be possible to add the support for literal type.

If I remember correctly `SPID.fromDefaults()` was planned to be a temporary helper method, which helps us migrate all  input format variants to use `SPID`. Especially there was need for `SPID.fromDefaults(streamId)` to help us to collect all "partition 0 is the default, if not specified otherwise" logic to one place (and later migrate those usages one by one).

@timoxley Do I remember correctly? Can we therefore drop `SPID.fromDefaults()` after we the migration?

Reply:
Created tickets:
- https://linear.app/streamr/issue/NET-566/client-use-spid
- https://linear.app/streamr/issue/NET-567/spidkey-as-template-literal-type